vta/hardware/chisel/src/main/scala/core/TensorGemm.scala

```diff
@@ -236,11 +254,14 @@ class TensorGemm(debug: Boolean = false)(implicit p: Parameters)
   when(state === sIdle) {
     inflight := 0.U
   }.elsewhen(!dec.reset) {
-    when(state === sReadTensor) { // issue a tensor
-      inflight := inflight + 1.U
-    }.elsewhen(mvc.io.acc_o.data.valid) { // commit a tensor
-      inflight := inflight - 1.U
-    }
+    when((state === sReadTensor) && mvc.io.acc_o.data.valid) { // issue & commit
```

Comment:
Is this a bug fix?

Reply:
Yes, if `latency=3` and we don't handle this case here, the hardware would wait infinitely long for `inflight` going to zero. But it never does, because sReadTensor state and `mvc.io.acc_o.data.valid` might simultaneously fire when latency equals to 3, and in that case, the increase amount in the `inflight` counter should be zero.

Reply:
Great, thanks for the catch.